### DISCUSSION TITLE
New algorithm includes training period?

### DISCUSSION THREAD

**Discussion by Jmuccigr** [retorquere/zotero-better-bibtex#3004] [Q&A] [answered]
I noticed that I have a couple of cite keys in my library that are somehow the same. They belong to a name that ends in "Jr." When I unfix the cite key, it gets redone with the trailing period, which the old cite keys don't have, so NameJr.2024 instead of NameJr2024. Now I can fix this manually, but I'm wondering whether this is a change in the algorithm or a result of me putting a final period after the "Jr" when it was lacking before.

**Comment by retorquere** [comment, marked as answer]
I don't recall a change in the algorithm that would be relevant, but

> or a result of me putting a final period after the "Jr" when it was lacking before.

this I cannot know of course

**Reply by Jmuccigr** [reply]
In the end I just fixed the two of them manually. I don't want to have to change all of these, so I'm just going to leave them alone.

PS **Not** including the period makes more sense to me.

PPS I've just discovered that the Zotero export to csv is very unreliable and left stuff out, whereas BBT's expert to json worked well.

**Reply by retorquere** [reply]
You can use `nopunct` to remove the period.

**Reply by retorquere** [reply]
> PS Not including the period makes more sense to me.

this is a very personal matter, trust me. Preferences on this kind of thing vary wildly.

**Reply by Jmuccigr** [reply]
I have no doubt! :-)

Thanks again for a great plugin.
